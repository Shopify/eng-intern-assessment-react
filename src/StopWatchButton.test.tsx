import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import StopwatchButton from './StopWatchButton';

describe('StopwatchButton component', () => {
  it('renders button with correct label', () => {
    const mockOnClick = jest.fn(); 
    const label = 'Start'; 
    const { getByText } = render(<StopwatchButton onClick={mockOnClick} label={label} />);
    
    // Find the button element by text 
    const buttonElement = getByText(label);
    
    // Ensure button is rendered
    expect(buttonElement).toBeDefined();
  });

  it('calls onClick function when button is clicked', () => {
    const mockOnClick = jest.fn(); // Create a mock function
    const label = 'Start'; // Label for the button
    const { getByText } = render(<StopwatchButton onClick={mockOnClick} label={label} />);
    
    // Find the button element by text 
    const buttonElement = getByText(label);
    
    // Simulate a click event on the button
    fireEvent.click(buttonElement);
    
    // Ensure onClick function is called
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
