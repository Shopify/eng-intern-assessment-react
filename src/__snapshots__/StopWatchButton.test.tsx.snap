// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopWatchButton tests render should match snapshot file 1`] = `
<button
  disabled={false}
  onClick={[MockFunction]}
>
  STOP
</button>
`;
